src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -172,65 +172,83 @@ namespace nupic {
               sp2.getMinPctActiveDutyCycles()));
 
 
-    Real boostFactors1[numColumns];
-    Real boostFactors2[numColumns];
+    auto boostFactors1 = new Real[numColumns];
```

Comment:
This changes the variable to a pointer, which is not what we want. Why are you making these changes?